pkg/server/api/github.go

```diff
@@ -30,14 +31,12 @@ func (c *Api) getClient() *githubv4.Client {
 	return githubv4.NewClient(&http.Client{Transport: itr})
 }
 
-func (c *Api) fetchUserMappings() (map[string]string, error) {
-	client := c.getClient()
-
+func (c *Api) fetchUserMappings(client *githubv4.Client) (map[string]string, error) {
```

Comment:
It feels somewhat inconsistent that `fetchUserMappings` now needs the GH client to be passed via args, but `updateUserMappings` does not. I see that this helps with writing tests, but could we maybe include the GH client in the `Api` struct? This would still let us mock the GH client will keeping the API cleaner.

I think in general this shows that maybe GH-related functionality should live in a sub-package and be properly instantiated, but probably we can refactor that in the future when there's a bigger need for it.

Reply:
Moving the GH related functionality would make it harder to test creating a certificate with GitHub options through an HTTP request. We would have to make a lot of functionality in the api package public to allow for that